admiral/pkg/controller/admiral/controller.go

```diff
@@ -102,7 +102,8 @@ func (c *Controller) Run(stopCh <-chan struct{}) {
 	}
 
 	log.Infof("Informer caches synced for controller=%v, current keys=%v", c.name, c.informer.GetStore().ListKeys())
-	wait.Until(c.runWorker, 5*time.Second, stopCh)
+	//wait for 30 seconds for the first time (for all caches to sync)
+	wait.Until(c.runWorker, 30 * time.Second, stopCh)
```

Comment:
Curious to know the reason to increase the sync time. I recall, there is a waitForSync or similarly named function given by k8s-api's client-go. Is it possible to use that for individual caches?

Reply:
The controller is already using `waitForCacheSync` before (lines 99-101) in this same file. This delay is an attempt to allow all controllers to complete sync given not all are initialized at the same time.